### PR TITLE
Add ancestor chain bit

### PR DESCRIPTION
**Pull request by aselya**

### CODE REVIEW

File: draft-cutler-httpbis-partitioned-cookies.md
Comment (DCtheTall) [inline draft-cutler-httpbis-partitioned-cookies.md:108]: This line's change is not needed since we already include the ancestor-chain bit in the definition of computing the partition key.

File: draft-cutler-httpbis-partitioned-cookies.md
Comment (DCtheTall) [inline draft-cutler-httpbis-partitioned-cookies.md:200]: Nit: trailing space.

File: draft-cutler-httpbis-partitioned-cookies.md
Comment (DCtheTall) [inline draft-cutler-httpbis-partitioned-cookies.md:70]: Paraphrasing from the [storage partitioning repo](https://github.com/privacycg/storage-partitioning#additional-keying) a bit here:

"In addition to double-keying cookies on top-level site, we propose to additionally key cookies on a bit indicating whether the frame tree includes a cross-site ancestor to prevent attacks between framed documents."

**Review by DCtheTall — CHANGES_REQUESTED** [review]
Looking better, still needs some changes before it is ready to merge.

**Comment by aselya** [comment]
Updated based on review.

File: draft-cutler-httpbis-partitioned-cookies.md
Comment (DCtheTall) [inline draft-cutler-httpbis-partitioned-cookies.md:91]: This is, admittedly, a bit of an informal definition of the ancestor-chain bit. Considering the target of this goal is to merge this into the cookie spec, RFC6265bis, perhaps we can reference this section and determine the boolean based on that result: https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-rfc6265bis-13#section-5.2

File: draft-cutler-httpbis-partitioned-cookies.md
Comment (DCtheTall) [inline draft-cutler-httpbis-partitioned-cookies.md:92]: Grammar here seems off. Perhaps something like "...cookie-partition-key's first element, the site (as defined in {{HTML}}) used for the key, is same-site with the top-document associated with the non-HTTP API."

**Review by DCtheTall — CHANGES_REQUESTED** [review]
Once these comments are addressed it is ready for merge.

File: draft-cutler-httpbis-partitioned-cookies.md
Comment (aselya) [reply draft-cutler-httpbis-partitioned-cookies.md]: Updated

File: draft-cutler-httpbis-partitioned-cookies.md
Comment (DCtheTall) [inline draft-cutler-httpbis-partitioned-cookies.md:91]: Nit: s/indicating if the site the cookie was set on/indicating if the context the cookie was created in.

**Review by DCtheTall — APPROVED** [review]
LGTM, one phrasing nit but not worth holding up for another review after :)

File: draft-cutler-httpbis-partitioned-cookies.md
Comment (aselya) [reply draft-cutler-httpbis-partitioned-cookies.md]: Done

**Comment by johannhof** [comment]
@DCtheTall are we good to merge this?

**Comment by wenglaurie30** [comment]
aselya:add-ancestor-chain-bit

File: draft-cutler-httpbis-partitioned-cookies.md
Comment (DCtheTall) [inline draft-cutler-httpbis-partitioned-cookies.md:91]: `and the whose second` should be `and whose second`

**Review by DCtheTall — APPROVED** [review]

File: draft-cutler-httpbis-partitioned-cookies.md
Comment (aselya) [reply draft-cutler-httpbis-partitioned-cookies.md]: Took a crack at it. Feel free to make edits

File: draft-cutler-httpbis-partitioned-cookies.md
Comment (DCtheTall) [inline draft-cutler-httpbis-partitioned-cookies.md:91]: Let's use an appositive here instead, and let's also keep it all one word in kebab-case.

`whose second element is a boolean, which we will call "cross-site-ancestor",`

File: draft-cutler-httpbis-partitioned-cookies.md
Comment (DCtheTall) [inline draft-cutler-httpbis-partitioned-cookies.md:93]: I am not sure this is entirely correct, what if there is a script in an A>B>A embedded frame?

**Review by DCtheTall — CHANGES_REQUESTED** [review]
Took another pass, it's coming together but there is one comment that is blocking merging this.

File: draft-cutler-httpbis-partitioned-cookies.md
Comment (aselya) [reply draft-cutler-httpbis-partitioned-cookies.md]: Removed the language in lieu of calling out that edge case specifically.

File: draft-cutler-httpbis-partitioned-cookies.md
Comment (DCtheTall) [inline draft-cutler-httpbis-partitioned-cookies.md:92]: Nit: spelling of "associated" is off.

Comment (DCtheTall) [inline draft-cutler-httpbis-partitioned-cookies.md:92]: I also do not think this definition covers A>B>A properly? We may want to defer to how RFC6265bis treats retrieving cookies from a non-HTTP API, https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-rfc6265bis-14#name-retrieval-model, namely that we use the document's URL and SiteForCookies to determine same- or cross-sitedness

Comment (DCtheTall) [inline draft-cutler-httpbis-partitioned-cookies.md:92]: I am not sure if this definition is correct for A>B>A either. We may want to defer to how RFC6265bis handles non-HTTP cookie retrieval, https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-rfc6265bis-14#section-5.7.2, where they suggest we use the document's URL and the document request's SiteForCookies to determine first- or third-partiness.

Comment (DCtheTall) [inline draft-cutler-httpbis-partitioned-cookies.md:92]: Nit: spelling of "associated" is off.

**Review by DCtheTall — CHANGES_REQUESTED** [review]

File: draft-cutler-httpbis-partitioned-cookies.md
Comment (aselya) [reply draft-cutler-httpbis-partitioned-cookies.md]: Removed

**Review by DCtheTall — APPROVED** [review]

**Pull request merged**
